LandSandBoat/server
# Add !gotonpc command

**_I affirm:_**

## What does this pull request do?

- Adds !gotonpc command used to go to an npc by name, in any zone
- Adds a set of `GetNPCsByName` methods to get npcs based on a search pattern
- Adds a "matches" function in utils.cpp to perform string search with wildcard character

## Steps to test these changes

on a GM character, type:
!gotonpc %anto%

Then type:
!gotonpc %anto% 2

Should teleport you to the chosen npc

![goto](https://user-images.githubusercontent.com/116968546/200108021-fb975c2a-6a55-4967-b69d-ab516ca9355b.png)
